### ISSUE TITLE
Cryogenics: Discuss Improvements for Support Monitoring

### ISSUE THREAD

**Issue by DominicOram** [ISISComputingGroup/IBEX#5369]
As a cryogenics support user I would like to be able to monitor arbitrary PVs. Currently I create my own OPIs in CSS to do this. A meeting should be arranged to discuss the following:
* The possibility of using Pheobus instead of CSS
* The request for a PV finder to be able to find e.g. all Triton PVs (Channel Finder could be used here)

## Acceptance Criteria
- A demo installation of Pheobus and Channel Finder is set up
- This is presented to cryogenics as a potential solution for feedback

**Comment by John-Holt-Tessella** [comment]
Can we make sure that this is the right decision before helping them towards this goal. Let us get some requirements for there use cases. 
If we recommend this course it means that we will be supporting Pheobus and channel finder; even if only lightly. Does it also mean that we can no longer arbitrarily change PV names without informing them (I would be cross if I had set up an display building screen only to have PVs change underneath me). Finally they would not get any improvements that IBEX Gui might bring them, there may be none for cryogenics (e.g. they may not have the log plotter right clicked from opis but maybe that is not important).

**Comment by ChrisM-S** [comment]
Yes I agree with @John-Holt-Tessella.  One alternative architecture might be leverage these sort of outputs from the instrument as data streams (via the data streaming infrastructure) which could be subscribed to by _any_ client - catching/finding PVs remotely and their  naming are quite fragile in that you have to catch them when they are there (which the instruments probably could do best locally via a hidden block subsequently forwarded to a published stream).
Requesting and displaying (including plotting) data from the published streams seems to be a job which could be done by other clients (possibly already existing? - e.g. Mantid?).  It might be better to still do the Phoebus work but focus it on evolving the IBEX client directly and keeping the focus on the IBEX instrument control client instead?

**Comment by DominicOram** [comment]
> Can we make sure that this is the right decision before helping them towards this goal. Let us get some requirements for there use cases.

I agree

> Does it also mean that we can no longer arbitrarily change PV names without informing them (I would be cross if I had set up an display building screen only to have PVs change underneath me). 

It's my understanding that they are already using vanilla CSS so they're already dependent on PV names

> Finally they would not get any improvements that IBEX Gui might bring them, there may be none for cryogenics (e.g. they may not have the log plotter right clicked from opis but maybe that is not important).

Graphing is absolutely possible in vanilla CSS/Pheobus. But yes, we should evaluate IBEX/CSS/Pheobus and the pros and cons of each.

> One alternative architecture might be leverage these sort of outputs from the instrument as data streams (via the data streaming infrastructure) which could be subscribed to by any client - catching/finding PVs remotely and their naming are quite fragile in that you have to catch them when they are there (which the instruments probably could do best locally via a hidden block subsequently forwarded to a published stream). Requesting and displaying (including plotting) data from the published streams seems to be a job which could be done by other clients (possibly already existing? - e.g. Mantid?)

I don't understand how this is different from EPICS, we can already have any client we want looking at PVs. We have a number of pre-existing plotting tools that were developed by the whole community.

I probably put the cart before the horse in writing this ticket, sorry, I'll change the acceptance criteria. But we shouldn't forget that they already have something that is working for them and requires very little maintenance from our side. Let's make sure we don't turn this into some huge project of re-writing parts of the system 'cos it will never get done.
